plugins/routers/joint.routers.manhattan.js

```diff
@@ -382,7 +382,31 @@ joint.routers.manhattan = (function() {
             .difference(excludedEnds)
             // remove all elements whose type is listed in excludedTypes array
             .reject(function(element) {
-                return _.contains(opt.excludeTypes, element.get('type'));
+                // reject any element which is an ancestor of either source or target
+                var excludeAncestors = [];
+
+                var findAncestors = function(cell, arr){
```

Comment:
Would be nice if we could have `getAncestors()` method on the `joint.dia.Cell` prototype itself. Returning an array of ancestor cells (or `[]` if the cell has no parent).

PS: I would rather use an iteration (`while (parent) { .. }`) instead of a recursion.